business_requirement_deliverable_default/models/business_requirement_deliverable_default.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+# © 2016 Elico Corp (https://www.elico-corp.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp import api, fields, models
+
+
+class BusinessRequirementDeliverable(models.Model):
+    _inherit = "business.requirement.deliverable"
+
+    def _prepare_resouce_lines(self):
+        rl_data = self.product_id.resource_lines.copy_data()
```

Comment:
@siyuancn Yes, but in that case it will return an empty list `[]` so nothing to do update in that case. It also tested because it's mostly the state for Products (I have to put some resources in one or two products, so it have resources and the others not).